saleor/graphql/core/fields.py

```diff
@@ -66,3 +68,57 @@ def resolve_connection(cls, connection, default_manager, args, iterable):
         connection.iterable = iterable
         connection.length = _len
         return connection
+
+
+class FilterInputConnectionField(DjangoConnectionField):
+    def __init__(self,  *args, **kwargs):
+        self.filter_field_name = kwargs.pop('filter_field_name', 'filter')
+        self.filter_input = kwargs.get(self.filter_field_name)
+        self.filterset_class = None
+        if self.filter_input:
+            self.filterset_class = self.filter_input.filterset_class
+        super().__init__(*args, **kwargs)
+
+    @classmethod
+    def connection_resolver(
+            cls, resolver, connection, default_manager, max_limit,
+            enforce_first_or_last, filterset_class, filters_name, root, info,
+            **args):
+
+        # Disable `enforce_first_or_last` if not querying for `edges`.
+        values = [
+            field.name.value
+            for field in info.field_asts[0].selection_set.selections]
+        if 'edges' not in values:
+            enforce_first_or_last = False
+
+        filter_input = args.get(filters_name)
+        qs = default_manager.get_queryset()
+        if filter_input and filterset_class:
+            qs = filterset_class(
+                data=dict(filter_input),
+                queryset=default_manager.get_queryset(),
+                request=info.context).qs
+
+        return super().connection_resolver(
+            resolver,
+            connection,
+            qs,
+            max_limit,
+            enforce_first_or_last,
+            root,
+            info,
+            **args
+        )
+
+    def get_resolver(self, parent_resolver):
+        return partial(
```

Comment:
If I understand correctly, you wanted to get the same `partial` as base `DjangoConnectionField`'s `get_resolver` returns extended by two arguments. In this case, I think you can do
```
returns partial(super().get_resolver(parent_resolver), self.filterset_class, self.filter_field_name)
```
as partial of partial, returns another partial. I am not 100% sure however, but if tests/usage won't fail, we could save some lines of code.

Reply:
Will test it. Thanks!